src/python/pants/core/subsystems/python_bootstrap.py

```diff
@@ -42,63 +36,48 @@ class PythonBootstrapSubsystem(Subsystem):
         """
     )
 
-    search_path = StrListOption(
-        default=["<PYENV>", "<PATH>"],
-        help=softwrap(
-            """
-            A list of paths to search for Python interpreters.
-
-            Which interpeters are actually used from these paths is context-specific:
-            the Python backend selects interpreters using options on the `python` subsystem,
-            in particular, the `[python].interpreter_constraints` option.
-
-            You can specify absolute paths to interpreter binaries
-            and/or to directories containing interpreter binaries. The order of entries does
-            not matter.
-
-            The following special strings are supported:
-
-              * `<PATH>`, the contents of the PATH env var
-              * `<ASDF>`, all Python versions currently configured by ASDF \
-                  `(asdf shell, ${HOME}/.tool-versions)`, with a fallback to all installed versions
-              * `<ASDF_LOCAL>`, the ASDF interpreter with the version in BUILD_ROOT/.tool-versions
-              * `<PYENV>`, all Python versions under $(pyenv root)/versions
-              * `<PYENV_LOCAL>`, the Pyenv interpreter with the version in BUILD_ROOT/.python-version
-              * `<PEXRC>`, paths in the PEX_PYTHON_PATH variable in /etc/pexrc or ~/.pexrc
-            """
-        ),
-        advanced=True,
-        metavar="<binary-paths>",
-        environment_sensitive=True,
-    )
-    names = StrListOption(
-        default=["python", "python3"],
-        help=softwrap(
-            """
-            The names of Python binaries to search for. See the `--search-path` option to
-            influence where interpreters are searched for.
-
-            This does not impact which Python interpreter is used to run your code, only what
-            is used to run internal tools.
-            """
-        ),
-        advanced=True,
-        metavar="<python-binary-names>",
-        environment_sensitive=True,
-    )
+    class EnvironmentAware(Subsystem.EnvironmentAware):
```

Comment:
Currently every `EnvironmentAware` needs to subclass `Subsystem.EnvironmentAware`.

Hopefully I can make that subclassing happen automagically, as this is a tiny bit tedious.